src/astro/slam_reduction.f90

```diff
@@ -3443,4 +3489,49 @@ subroutine eci2teme_rva(   this,    &
 
   end subroutine eci2teme_rva
 
+  !-----------------------------------------------------------------------------------------------
+    !
+    !> @anchor      getDeltaTime
+    !!
+    !> @brief       Get the delta time UT1-UTC for a given date
+    !> @author      Andrea Turchi
+    !!
+    !> @date        <ul>
+    !!                <li> 10.02.2023: initial design</li>
+    !!              </ul>
+    !> @param[in]   time_mjd      ! current MJD for requested delta_t
+    !!
+    !-----------------------------------------------------------------------------------------------
+  function getDeltaTime(this, time_mjd)
+
+    class(Reduction_type) :: this
+    real(dp)              :: getDeltaTime
+    real(dp), intent(in)  :: time_mjd
+
+    integer     :: idx                                                      ! index in EOP data array
+    type(eop_t) :: eop_intp
+
+    idx = int(time_mjd - eop_data(1)%mjd) + 1
+
+    ! if (time_mjd <= this%eop_last_obs_date) then
+    !     call this%interpolateEOP(idx, mod(time_mjd,1.d0), eop_intp)
+    ! else
+    !     call this%getEOPfromNGA(time_mjd, eop_intp)
+    ! end if
+    if (time_mjd > this%eop_last_obs_date .and. this%flag_nga_coeff) then
+      call this%getEOPfromNGA(time_mjd, eop_intp)
+    else
+      call this%interpolateEOP(idx, mod(time_mjd,1.d0), eop_intp)
+    end if
+
+    getDeltaTime = eop_data(idx)%dut1  ! already in seconds
+
+    if (eop_data(idx)%dut1 == 0.d0) then
+      write(*,*) "WARNING: after END PREDICTED a null value is returned!"
```

Comment:
Can you use the message system instead of WRITE statements?
Something like this:
`call slam_message('WARNING: after END PREDICTED a null value is returned',  LOG_AND_STDOUT)`